### PR TITLE
olaris-server: init at 0.4.0

### LINKED ISSUE #114305: package request: olaris-server
**Project description**

https://gitlab.com/olaris/olaris-server
https://gitlab.com/olaris/olaris-react

Olaris is an open-source, community driven, media manager and transcoding server. The main interface is the olaris-react project although in due time we hope to support multiple clients / applications.

It looks like `olaris-react` can be built and then the `build` dir can be copied into `olaris-server` and then olaris server bundles in the static resources. Hopefully this means that one package can be built to handle react, and the other can just use `buildGoModule` nicely.

**Metadata**

* homepage URL: https://gitlab.com/olaris/olaris-server
* source URL: https://gitlab.com/olaris/olaris-server
* license: gpl3
* platforms:  linux

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

Fixes #114305

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 203710` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>olaris-server</li>
</ul>
</details>

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 203710` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>olaris-server</li>
</ul>
</details>

File: pkgs/servers/olaris/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    hash = "sha256-iworyQqyTabTI0NpZHTdUBGZSCaiC5Dhr69mRtsHLOs=";
+  };
+
+  preBuild = ''
+    # cannot build olaris-react https://github.com/NixOS/nixpkgs/issues/203708
+    cp -r ${fetchzip {
+      url = "https://gitlab.com/api/v4/projects/olaris%2Folaris-react/jobs/artifacts/v${version}/download?job=build";
+      extension = "zip";
+      hash = "sha256-MkxBf/mGvtiOu0e79bMpd9Z/D0eOxhzPE+bKic//viM=";
+    }} react/build
```
Comment (SuperSandro2000) [inline pkgs/servers/olaris/default.nix:20]: Please don't inline this and move it into a let in

File: pkgs/servers/olaris/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    installShellCompletion --cmd olaris-server \
+      --bash <($out/bin/olaris-server completion bash) \
+      --fish <($out/bin/olaris-server completion fish) \
+      --zsh <($out/bin/olaris-server completion zsh)
+  '';
+
+
+  meta = with lib; {
+    homepage = "https://gitlab.com/olaris/olaris-server";
+    description = "A media manager and transcoding server.";
```
Comment (SuperSandro2000) [inline pkgs/servers/olaris/default.nix:48]: normally we would swap homepage and description

File: pkgs/servers/olaris/default.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  preBuild = let
+    olaris-react = fetchzip {
+      url = "https://gitlab.com/api/v4/projects/olaris%2Folaris-react/jobs/artifacts/v${version}/download?job=build";
+      extension = "zip";
+      hash = "sha256-MkxBf/mGvtiOu0e79bMpd9Z/D0eOxhzPE+bKic//viM=";
+    };
+  in ''
+    # cannot build olaris-react https://github.com/NixOS/nixpkgs/issues/203708
+    cp -r ${olaris-react} react/build
+    ls -l react
```
Comment (SuperSandro2000) [inline pkgs/servers/olaris/default.nix:23]:
```suggestion
```

File: pkgs/servers/olaris/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  preBuild = let
+    olaris-react = fetchzip {
+      url = "https://gitlab.com/api/v4/projects/olaris%2Folaris-react/jobs/artifacts/v${version}/download?job=build";
+      extension = "zip";
+      hash = "sha256-MkxBf/mGvtiOu0e79bMpd9Z/D0eOxhzPE+bKic//viM=";
+    };
+  in ''
+    # cannot build olaris-react https://github.com/NixOS/nixpkgs/issues/203708
+    cp -r ${olaris-react} react/build
```
Comment (SuperSandro2000) [inline pkgs/servers/olaris/default.nix:22]: Using a symlink wouldn't work here?

File: pkgs/servers/olaris/default.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+  postInstall = ''
+    installShellCompletion --cmd olaris-server \
+      --bash <($out/bin/olaris-server completion bash) \
+      --fish <($out/bin/olaris-server completion fish) \
+      --zsh <($out/bin/olaris-server completion zsh)
+  '';
+
+
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/servers/olaris/default.nix:48]:
```suggestion
  '';

  meta = with lib; {
```

File: pkgs/servers/olaris/default.nix (hunk 4 above)
Comment (urandom2) [reply pkgs/servers/olaris/default.nix:22]: correct, go:embed does not traverse symlinks: golang/go#44507

**Pull request merged**